File: frontend/src/types/deal.ts
export interface Store {
  id: number;
  name: string;
  slug: string;
  icon_url?: string;
  is_active: boolean;
}

export interface GamePrice {
  id: number;
  store_id: number;
  store_name: string;
  store_slug: string;
  store_icon?: string;
  country: string;
  currency: string;
  price: number;
  original_price: number;
  discount_percent: number;
  deal_url: string;
  is_best_deal: boolean;
}

export interface PriceHistoryPoint {
  price: number;
  original_price: number;
  discount_percent: number;
  recorded_at: string;
  store_name: string;
}

export interface HistoricalLow {
  lowest_price: number;
  highest_discount: number;
  lowest_price_date: string;
  store_name: string;
}

export interface GamePriceHistoryResponse {
  currency: string;
  historical_low?: HistoricalLow;
  history_points: PriceHistoryPoint[];
}

export interface DealSummary {
  id: number;
  game_id: number;
  game_slug: string;
  game_title: string;
  cover_image?: string;
  genres: string[];
  store_name: string;
  store_slug: string;
  store_icon?: string;
  country: string;
  currency: string;
  price: number;
  original_price: number;
  discount_percent: number;
  deal_url: string;
  is_free: boolean;
}

export interface DealListResponse {
  items: DealSummary[];
  total: number;
  page: number;
  page_size: number;
  currency: string;
}

export interface DealFilterParams {
  currency?: string;
  store_slug?: string;
  min_discount?: number;
  ordering?: 'discount' | 'price_low' | 'price_high';
  page?: number;
  page_size?: number;
}

File: backend/app/schemas/deal.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class StoreSchema(BaseModel):
    id: int
    name: str
    slug: str
    icon_url: Optional[str] = None
    is_active: bool = True

    class Config:
        from_attributes = True


class GamePriceSchema(BaseModel):
    id: int
    store_id: int
    store_name: str
    store_slug: str
    store_icon: Optional[str] = None
    country: str
    currency: str
    price: float
    original_price: float
    discount_percent: float
    deal_url: str
    is_best_deal: bool = False

    class Config:
        from_attributes = True


class PriceHistoryPoint(BaseModel):
    price: float
    original_price: float
    discount_percent: float
    recorded_at: datetime
    store_name: str

    class Config:
        from_attributes = True


class HistoricalLowSchema(BaseModel):
    lowest_price: float
    highest_discount: float
    lowest_price_date: datetime
    store_name: str


class GamePriceHistoryResponse(BaseModel):
    currency: str
    historical_low: Optional[HistoricalLowSchema] = None
    history_points: List[PriceHistoryPoint] = Field(default_factory=list)


class DealSummarySchema(BaseModel):
    id: int
    game_id: int
    game_slug: str
    game_title: str
    cover_image: Optional[str] = None
    genres: List[str] = Field(default_factory=list)
    store_name: str
    store_slug: str
    store_icon: Optional[str] = None
    country: str
    currency: str
    price: float
    original_price: float
    discount_percent: float
    deal_url: str
    is_free: bool = False

    class Config:
        from_attributes = True


class DealListResponse(BaseModel):
    items: List[DealSummarySchema]
    total: int
    page: int
    page_size: int
    currency: str
